csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -718,6 +711,8 @@ public void AddDefault(string facet, IObject servant)
             Protocol = protocol;
             IncomingFrameMaxSize = communicator.IncomingFrameMaxSize;
             AcceptNonSecure = communicator.AcceptNonSecure;
+
+            _dispatchInterceptors = Communicator.DefaultDispatchInterceptors.ToImmutableList();
```

Comment:
If `_dispatchInterceptors` was an `IReadOnlyList` we wouldn't have to call `ToImmutableList` (which might create new list?).

Reply:
It's like a cast since the backing field for Communicator.DefaultDispatchInterceptors is an ImmutableList<>.